### PR TITLE
Minor Updates

### PR DESCRIPTION
**Pull request by gamaanderson**
Updates to address issues #110, #108, #96 and test for dependencies in `setup.py` (#109)

### CODE REVIEW

File: artview/components/plot_grid.py [hunk 1]
```diff
@@ -254,11 +254,11 @@ def _fillLevelBox(self):
...
         # Loop through and create each level button
         if self.plot_type == "gridZ":
-            levels = self.Vgrid.value.axes['z_disp']['data']
+            levels = self.Vgrid.value.z['data']
         elif self.plot_type == "gridY":
-            levels = self.Vgrid.value.axes['y_disp']['data']
+            levels = self.Vgrid.value.y['data']
         elif self.plot_type == "gridX":
-            levels = self.Vgrid.value.axes['x_disp']['data']
+            levels = self.Vgrid.value.x['data']
```
Comment (nguy) [inline artview/components/plot_grid.py]: I haven't followed the GridDisplay updates closely. Is this change backwards compatible? Meaning does ```self.Vgrid.value.z``` map to ```self.Vgrid.value.axes['z_disp']``` in PyART or somewhere in ARTView?

Comment (nguy) [inline artview/components/plot_grid.py]: Wait now looking further below, either plot_grid or plot_grid_legacy is called within ARTView depending on version number, correct?

File: artview/components/plot_radar.py [hunk 2]
```diff
@@ -602,11 +602,16 @@ def toolZoomPanCmd(self):
...
+        if self.plot_type == 'radarPpi':
+            from .tools import ValueClick
+            self.tools['valueclick'] = ValueClick(
+                self.Vradar, self.Vtilt, self.Vfield,
+                self.units, self.ax, self.statusbar, parent=self.parent)
+            self.tools['valueclick'].connect()
+        else:
+            import warnings
+            warnings.warn("valueclick not implemented for airborne or rhi",
+                          NotImplementedError)
```
Comment (nguy) [inline artview/components/plot_radar.py]: I don't recall ValueClick not working with RHI/Airborne. Maybe I never tested....

File: artview/components/tools.py [hunk 3]
```diff
@@ -58,6 +60,8 @@ def reset_tools(tooldict):
     -----
     Disconnects all tools.
     '''
+    import warnings
+    warnings.warn("this function is deprecated")
     for tool in tooldict:
         if tooldict[tool] is not None:
             tooldict[tool].disconnect()
```
Comment (nguy) [inline artview/components/tools.py]: Just curious: what's the reason for putting the tool disconnects in the display scripts versus in the tools.py?

File: artview/plugins/io.py [hunk 4]
```diff
@@ -17,20 +17,30 @@
 
 from ..core import Component, Variable, common, QtGui, QtCore, componentsList
 
+# get list of read functions
+import inspect
+aux_read_functions = inspect.getmembers(pyart.aux_io, inspect.isfunction)
+read_functions = [
+    pyart.io.read,
+    pyart.io.read_grid,
+    pyart.io.read_grid_mdv] + [a[1] for a in aux_read_functions]
```
Comment (nguy) [inline artview/plugins/io.py]: This is pretty slick! I've never used inspect. Is it a built in Python library?

File: artview/plugins/io.py [hunk 5]
```diff
@@ -17,20 +17,30 @@
...
+try:
+    for func in read_functions:
+        try:
+            func(None)
+        except pyart.exceptions.MissingOptionalDependency:
+            brocken_read_functions.append(func)
+        except:
+            pass
+except:
+    pass
```
Comment (nguy) [inline artview/plugins/io.py]: Have you run into trouble with broken functions in Py-ART?

File: artview/components/plot_grid.py (hunk 1 above)
Comment (gamaanderson) [inline artview/components/plot_grid.py]: yes!

File: artview/components/tools.py (hunk 3 above)
Comment (gamaanderson) [inline artview/components/tools.py]: this is just a warning that this function is no longer being used.

I made however the opposite question, why remove from display class control over an internal part of it. There is no abstract concept of tool, it is an empiric construction, therefore any code separation is only visual.

File: artview/plugins/io.py (hunk 5 above)
Comment (gamaanderson) [inline artview/plugins/io.py]: `MissingOptionalDependency`

File: artview/components/tools.py (hunk 3 above)
Comment (nguy) [inline artview/components/tools.py]: Fair enough. I have no problems with deprecating this. And you are right, this probably should be internal rather calling out to another location.

File: artview/plugins/io.py (hunk 5 above)
Comment (nguy) [inline artview/plugins/io.py]: Ah, like h5py or such. Got it, nice!

File: artview/plugins/io.py (hunk 4 above)
Comment (gamaanderson) [inline artview/plugins/io.py]: Yes, it is python built in. This approach has suggested by @jjhelmus

**Comment by nguy** [comment]
So, I'm unable to open the gridded MDV file in the PyART testing directory. Would you mind sharing a gridded file with me to look at and make sure I can open?

**Comment by gamaanderson** [comment]
The pyart example should work, but they do not open with `read_grid`, only with `read_grid_mdv`. You need to use the open options of `FileList`. If it still don't work, I will pass you one of my files.

**Comment by nguy** [comment]
I tried that and nothing happens when selected ```read_grid_mdv``` for me.

**Comment by nguy** [comment]
Thanks for the file. I'm ready to merge unless you have anything else.

**Comment by gamaanderson** [comment]
No, that is all for this PR.
The following issues can be closed as they are solved or out-dated: #74 #79 #80 #96 #99 #108 #109 #110 (I can't close because I don't have the rights)

**Comment by nguy** [comment]
Thanks for the great work @gamaanderson !

**Pull request merged**